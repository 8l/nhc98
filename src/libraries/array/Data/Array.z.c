#include "newmacros.h"
#include "runtime.h"

#define ST_v205	((void*)startLabel+0)

static Node startLabel[] = {
 };
Node PM_Data_46Array[] = {
 	/* ST_v205: (byte 0) */
  bytes2word(68,97,116,97)
, bytes2word(46,65,114,114)
, bytes2word(97,121,0,0)
,};
