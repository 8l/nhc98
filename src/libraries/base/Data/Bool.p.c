#include "newmacros.h"
#include "runtime.h"

#define ST_v159	((void*)startLabel+0)

static Node startLabel[] = {
 };
Node PM_Data_46Bool[] = {
 	/* ST_v159: (byte 0) */
  bytes2word(68,97,116,97)
, bytes2word(46,66,111,111)
, bytes2word(108,0,0,0)
,};
