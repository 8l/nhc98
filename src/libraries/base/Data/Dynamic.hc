#include "newmacros.h"
#include "runtime.h"

#define C0_Data_46Dynamic_46Obj	((void*)startLabel+4)
#define CT_v467	((void*)startLabel+28)
#define v473	((void*)startLabel+78)
#define v474	((void*)startLabel+83)
#define CT_v478	((void*)startLabel+124)
#define v480	((void*)startLabel+174)
#define v481	((void*)startLabel+217)
#define CT_v483	((void*)startLabel+228)
#define FN_LAMBDA460	((void*)startLabel+276)
#define CT_v485	((void*)startLabel+296)
#define CF_LAMBDA460	((void*)startLabel+304)
#define FN_LAMBDA459	((void*)startLabel+316)
#define CT_v487	((void*)startLabel+336)
#define CF_LAMBDA459	((void*)startLabel+344)
#define FN_LAMBDA458	((void*)startLabel+356)
#define CT_v489	((void*)startLabel+376)
#define CF_LAMBDA458	((void*)startLabel+384)
#define v492	((void*)startLabel+438)
#define v494	((void*)startLabel+449)
#define CT_v497	((void*)startLabel+472)
#define FN_LAMBDA461	((void*)startLabel+520)
#define CT_v499	((void*)startLabel+540)
#define CF_LAMBDA461	((void*)startLabel+548)
#define v504	((void*)startLabel+594)
#define v506	((void*)startLabel+603)
#define v500	((void*)startLabel+608)
#define CT_v509	((void*)startLabel+624)
#define FN_LAMBDA462	((void*)startLabel+672)
#define CT_v511	((void*)startLabel+692)
#define CF_LAMBDA462	((void*)startLabel+700)
#define CT_v512	((void*)startLabel+752)
#define CT_v516	((void*)startLabel+836)
#define FN_LAMBDA464	((void*)startLabel+880)
#define CT_v518	((void*)startLabel+900)
#define CF_LAMBDA464	((void*)startLabel+908)
#define FN_LAMBDA463	((void*)startLabel+920)
#define CT_v520	((void*)startLabel+940)
#define CF_LAMBDA463	((void*)startLabel+948)
#define CT_v521	((void*)startLabel+972)
#define CT_v522	((void*)startLabel+1012)
#define CT_v523	((void*)startLabel+1052)
#define FN_Data_46Dynamic_46dynamicTc	((void*)startLabel+1080)
#define CT_v524	((void*)startLabel+1092)
#define CF_Data_46Dynamic_46dynamicTc	((void*)startLabel+1100)
#define FN_LAMBDA465	((void*)startLabel+1120)
#define CT_v526	((void*)startLabel+1140)
#define CF_LAMBDA465	((void*)startLabel+1148)
#define CT_v527	((void*)startLabel+1184)
#define CT_v528	((void*)startLabel+1232)
#define CT_v529	((void*)startLabel+1280)
#define ST_v484	((void*)startLabel+1308)
#define ST_v519	((void*)startLabel+1322)
#define ST_v517	((void*)startLabel+1325)
#define ST_v486	((void*)startLabel+1328)
#define ST_v498	((void*)startLabel+1350)
#define ST_v510	((void*)startLabel+1424)
#define ST_v525	((void*)startLabel+1489)
#define ST_v488	((void*)startLabel+1497)
extern Node FN_Data_46Typeable_46funResultTy[];
extern Node FN_NonStdUnsafeCoerce_46unsafeCoerce[];
extern Node FN_NHC_46Internal_46_95apply1[];
extern Node FN_Prelude_46Show_46Data_46Dynamic_46Dynamic_46show[];
extern Node FN_Prelude_46_43_43[];
extern Node FN_Prelude_46error[];
extern Node FN_Data_46Typeable_46typeOf[];
extern Node FN_Prelude_46Eq_46Data_46Typeable_46TypeRep_46_61_61[];
extern Node CF_Prelude_46otherwise[];
extern Node FN_NHC_46Internal_46_95patternMatchFail[];
extern Node FN_Prelude_46showString[];
extern Node FN_Prelude_46Show_46Data_46Typeable_46TypeRep_46showsPrec[];
extern Node FN_Prelude_46_46[];
extern Node FN_Prelude_46_95_46show[];
extern Node CF_Prelude_46Show_46Data_46Dynamic_46Dynamic[];
extern Node FN_Prelude_46_95_46showsType[];
extern Node FN_Prelude_46_95_46showList[];
extern Node FN_Data_46Typeable_46mkTyCon[];
extern Node FN_Data_46Typeable_46mkTyConApp[];

static Node startLabel[] = {
  42
,	/* C0_Data_46Dynamic_46Obj: (byte 0) */
  CONSTR(0,0,0)
, bytes2word(1,0,0,1)
, useLabel(CT_v467)
,};
Node FN_Data_46Dynamic_46dynTypeRep[] = {
  bytes2word(NEEDSTACK_I16,SELECTOR_EVAL,SELECT,1)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
,	/* CT_v467: (byte 0) */
  HW(0,1)
, 0
,};
Node F0_Data_46Dynamic_46dynTypeRep[] = {
  CAPTAG(useLabel(FN_Data_46Dynamic_46dynTypeRep),1)
, bytes2word(0,0,2,0)
, bytes2word(1,1,0,2)
, useLabel(CT_v478)
,};
Node FN_Data_46Dynamic_46dynApply[] = {
  bytes2word(NEEDSTACK_I16,PUSH_ZAP_ARG_I1,EVAL,UNPACK)
, bytes2word(2,PUSH_ZAP_ARG_I2,EVAL,NEEDHEAP_I32)
, bytes2word(UNPACK,2,PUSH_HEAP,HEAP_CVAL_I3)
, bytes2word(HEAP_P1,3,HEAP_I1,ZAP_STACK_P1)
, bytes2word(3,EVAL,NEEDHEAP_I32,TABLESWITCH)
, bytes2word(2,NOP,TOP(4),BOT(4))
,	/* v473: (byte 2) */
  bytes2word(TOP(9),BOT(9),POP_I1,PUSH_HEAP)
,	/* v474: (byte 3) */
  bytes2word(HEAP_CVAL_N1,1,RETURN,UNPACK)
, bytes2word(1,HEAP_CVAL_I4,HEAP_P1,4)
, bytes2word(HEAP_CVAL_I5,HEAP_OFF_N1,3,HEAP_I2)
, bytes2word(HEAP_CVAL_N1,2,HEAP_P1,0)
, bytes2word(HEAP_OFF_N1,5,PUSH_HEAP,HEAP_CVAL_IN3)
, bytes2word(HEAP_OFF_N1,4,RETURN,ENDCODE)
, bytes2word(0,0,0,0)
, 0
, CONSTR(1,1,0)
, CONSTR(0,2,0)
, CONSTR(0,0,0)
,	/* CT_v478: (byte 0) */
  HW(3,2)
, 0
,};
Node F0_Data_46Dynamic_46dynApply[] = {
  CAPTAG(useLabel(FN_Data_46Dynamic_46dynApply),2)
, VAPTAG(useLabel(FN_Data_46Typeable_46funResultTy))
, VAPTAG(useLabel(FN_NonStdUnsafeCoerce_46unsafeCoerce))
, VAPTAG(useLabel(FN_NHC_46Internal_46_95apply1))
, bytes2word(0,0,2,0)
, bytes2word(1,1,0,2)
, useLabel(CT_v483)
,};
Node FN_Data_46Dynamic_46dynApp[] = {
  bytes2word(NEEDHEAP_I32,PUSH_HEAP,HEAP_CVAL_I3,HEAP_ARG_ARG)
, bytes2word(1,2,EVAL,NEEDHEAP_I32)
, bytes2word(TABLESWITCH,2,TOP(4),BOT(4))
,	/* v480: (byte 2) */
  bytes2word(TOP(47),BOT(47),POP_I1,HEAP_CVAL_I4)
, bytes2word(HEAP_CVAL_I5,HEAP_CVAL_P1,6,HEAP_ARG)
, bytes2word(1,HEAP_CVAL_P1,7,HEAP_CVAL_P1)
, bytes2word(6,HEAP_ARG,2,HEAP_CVAL_P1)
, bytes2word(8,HEAP_OFF_N1,4,HEAP_OFF_N1)
, bytes2word(4,HEAP_CVAL_P1,8,HEAP_OFF_N1)
, bytes2word(9,HEAP_OFF_N1,5,HEAP_CVAL_P1)
, bytes2word(8,HEAP_OFF_N1,13,HEAP_OFF_N1)
, bytes2word(5,HEAP_CVAL_P1,8,HEAP_OFF_N1)
, bytes2word(17,HEAP_OFF_N1,5,PUSH_HEAP)
, bytes2word(HEAP_CVAL_P1,9,HEAP_OFF_N1,4)
,	/* v481: (byte 1) */
  bytes2word(RETURN_EVAL,UNPACK,1,PUSH_P1)
, bytes2word(0,RETURN_EVAL,ENDCODE,0)
, bytes2word(0,0,0,0)
,	/* CT_v483: (byte 0) */
  HW(7,2)
, 0
,};
Node F0_Data_46Dynamic_46dynApp[] = {
  CAPTAG(useLabel(FN_Data_46Dynamic_46dynApp),2)
, VAPTAG(useLabel(FN_Data_46Dynamic_46dynApply))
, VAPTAG(useLabel(FN_LAMBDA458))
, VAPTAG(useLabel(FN_LAMBDA459))
, VAPTAG(useLabel(FN_Prelude_46Show_46Data_46Dynamic_46Dynamic_46show))
, VAPTAG(useLabel(FN_LAMBDA460))
, VAPTAG(useLabel(FN_Prelude_46_43_43))
, VAPTAG(useLabel(FN_Prelude_46error))
, bytes2word(0,0,0,0)
, useLabel(CT_v485)
,	/* FN_LAMBDA460: (byte 0) */
  bytes2word(NEEDHEAP_I32,PUSH_CADR_N1,2,STRING)
, bytes2word(RETURN,ENDCODE,0,0)
, bytes2word(0,0,0,0)
, CONSTRW(1,0)
, useLabel(ST_v484)
,	/* CT_v485: (byte 0) */
  HW(0,0)
, 0
,	/* CF_LAMBDA460: (byte 0) */
  VAPTAG(useLabel(FN_LAMBDA460))
, bytes2word(0,0,0,0)
, useLabel(CT_v487)
,	/* FN_LAMBDA459: (byte 0) */
  bytes2word(NEEDHEAP_I32,PUSH_CADR_N1,2,STRING)
, bytes2word(RETURN,ENDCODE,0,0)
, bytes2word(0,0,0,0)
, CONSTRW(1,0)
, useLabel(ST_v486)
,	/* CT_v487: (byte 0) */
  HW(0,0)
, 0
,	/* CF_LAMBDA459: (byte 0) */
  VAPTAG(useLabel(FN_LAMBDA459))
, bytes2word(0,0,0,0)
, useLabel(CT_v489)
,	/* FN_LAMBDA458: (byte 0) */
  bytes2word(NEEDHEAP_I32,PUSH_CADR_N1,2,STRING)
, bytes2word(RETURN,ENDCODE,0,0)
, bytes2word(0,0,0,0)
, CONSTRW(1,0)
, useLabel(ST_v488)
,	/* CT_v489: (byte 0) */
  HW(0,0)
, 0
,	/* CF_LAMBDA458: (byte 0) */
  VAPTAG(useLabel(FN_LAMBDA458))
, bytes2word(0,0,2,0)
, bytes2word(1,1,0,2)
, useLabel(CT_v497)
,};
Node FN_Data_46Dynamic_46fromDynamic[] = {
  bytes2word(NEEDSTACK_I16,PUSH_ZAP_ARG_I2,EVAL,NEEDHEAP_I32)
, bytes2word(UNPACK,2,PUSH_HEAP,HEAP_CVAL_I3)
, bytes2word(HEAP_I2,HEAP_CVAL_I4,HEAP_ARG,1)
, bytes2word(HEAP_CVAL_I5,HEAP_OFF_N1,3,HEAP_P1)
, bytes2word(0,PUSH_HEAP,HEAP_CVAL_P1,6)
, bytes2word(HEAP_I2,HEAP_OFF_N1,5,ZAP_ARG_I1)
, bytes2word(ZAP_STACK_P1,3,ZAP_STACK_P1,2)
, bytes2word(EVAL,NEEDHEAP_I32,JUMPFALSE,7)
, bytes2word(0,PUSH_HEAP,HEAP_CVAL_N1,1)
,	/* v492: (byte 2) */
  bytes2word(HEAP_I1,RETURN,PUSH_CVAL_P1,7)
, bytes2word(EVAL,NEEDHEAP_I32,JUMPFALSE,6)
, bytes2word(0,PUSH_HEAP,HEAP_CVAL_N1,2)
,	/* v494: (byte 1) */
  bytes2word(RETURN,HEAP_CVAL_P1,8,PUSH_HEAP)
, bytes2word(HEAP_CVAL_P1,9,HEAP_OFF_N1,2)
, bytes2word(RETURN_EVAL,ENDCODE,0,0)
, bytes2word(0,0,0,0)
, CONSTR(0,0,0)
, CONSTR(1,1,0)
,	/* CT_v497: (byte 0) */
  HW(7,2)
, 0
,};
Node F0_Data_46Dynamic_46fromDynamic[] = {
  CAPTAG(useLabel(FN_Data_46Dynamic_46fromDynamic),2)
, VAPTAG(useLabel(FN_NonStdUnsafeCoerce_46unsafeCoerce))
, VAPTAG(useLabel(FN_Data_46Typeable_46typeOf))
, VAPTAG(useLabel(FN_NHC_46Internal_46_95apply1))
, VAPTAG(useLabel(FN_Prelude_46Eq_46Data_46Typeable_46TypeRep_46_61_61))
, useLabel(CF_Prelude_46otherwise)
, VAPTAG(useLabel(FN_LAMBDA461))
, VAPTAG(useLabel(FN_NHC_46Internal_46_95patternMatchFail))
, bytes2word(0,0,0,0)
, useLabel(CT_v499)
,	/* FN_LAMBDA461: (byte 0) */
  bytes2word(NEEDHEAP_I32,PUSH_CADR_N1,2,STRING)
, bytes2word(RETURN,ENDCODE,0,0)
, bytes2word(0,0,0,0)
, CONSTRW(1,0)
, useLabel(ST_v498)
,	/* CT_v499: (byte 0) */
  HW(0,0)
, 0
,	/* CF_LAMBDA461: (byte 0) */
  VAPTAG(useLabel(FN_LAMBDA461))
, bytes2word(3,0,2,1)
, bytes2word(1,2,0,3)
, useLabel(CT_v509)
,};
Node FN_Data_46Dynamic_46fromDyn[] = {
  bytes2word(NEEDSTACK_I16,PUSH_ZAP_ARG_I2,EVAL,NEEDHEAP_I32)
, bytes2word(UNPACK,2,HEAP_CVAL_I3,HEAP_ARG)
, bytes2word(1,HEAP_CVAL_I4,HEAP_OFF_N1,3)
, bytes2word(HEAP_ARG,3,PUSH_HEAP,HEAP_CVAL_I5)
, bytes2word(HEAP_OFF_N1,4,HEAP_I1,ZAP_ARG_I1)
, bytes2word(EVAL,NEEDHEAP_I32,JUMPFALSE,7)
, bytes2word(0,PUSH_HEAP,HEAP_CVAL_P1,6)
,	/* v504: (byte 2) */
  bytes2word(HEAP_I2,RETURN_EVAL,PUSH_CVAL_P1,7)
, bytes2word(EVAL,NEEDHEAP_I32,JUMPFALSE,4)
,	/* v506: (byte 3) */
  bytes2word(0,PUSH_ARG_I3,RETURN_EVAL,POP_P1)
,	/* v500: (byte 4) */
  bytes2word(2,JUMP,2,0)
, bytes2word(HEAP_CVAL_P1,8,PUSH_HEAP,HEAP_CVAL_P1)
, bytes2word(9,HEAP_OFF_N1,2,RETURN_EVAL)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
,	/* CT_v509: (byte 0) */
  HW(7,3)
, 0
,};
Node F0_Data_46Dynamic_46fromDyn[] = {
  CAPTAG(useLabel(FN_Data_46Dynamic_46fromDyn),3)
, VAPTAG(useLabel(FN_Data_46Typeable_46typeOf))
, VAPTAG(useLabel(FN_NHC_46Internal_46_95apply1))
, VAPTAG(useLabel(FN_Prelude_46Eq_46Data_46Typeable_46TypeRep_46_61_61))
, VAPTAG(useLabel(FN_NonStdUnsafeCoerce_46unsafeCoerce))
, useLabel(CF_Prelude_46otherwise)
, VAPTAG(useLabel(FN_LAMBDA462))
, VAPTAG(useLabel(FN_NHC_46Internal_46_95patternMatchFail))
, bytes2word(0,0,0,0)
, useLabel(CT_v511)
,	/* FN_LAMBDA462: (byte 0) */
  bytes2word(NEEDHEAP_I32,PUSH_CADR_N1,2,STRING)
, bytes2word(RETURN,ENDCODE,0,0)
, bytes2word(0,0,0,0)
, CONSTRW(1,0)
, useLabel(ST_v510)
,	/* CT_v511: (byte 0) */
  HW(0,0)
, 0
,	/* CF_LAMBDA462: (byte 0) */
  VAPTAG(useLabel(FN_LAMBDA462))
, bytes2word(0,0,2,0)
, bytes2word(1,1,0,2)
, useLabel(CT_v512)
,};
Node FN_Data_46Dynamic_46toDyn[] = {
  bytes2word(NEEDHEAP_I32,HEAP_CVAL_I3,HEAP_ARG,1)
, bytes2word(HEAP_CVAL_I4,HEAP_OFF_N1,3,HEAP_ARG)
, bytes2word(2,HEAP_CVAL_I5,HEAP_ARG,2)
, bytes2word(PUSH_HEAP,HEAP_CVAL_N1,1,HEAP_OFF_N1)
, bytes2word(6,HEAP_OFF_N1,4,RETURN)
, bytes2word(ENDCODE,0,0,0)
, bytes2word(0,0,0,0)
, 0
, CONSTR(0,2,0)
,	/* CT_v512: (byte 0) */
  HW(3,2)
, 0
,};
Node F0_Data_46Dynamic_46toDyn[] = {
  CAPTAG(useLabel(FN_Data_46Dynamic_46toDyn),2)
, VAPTAG(useLabel(FN_Data_46Typeable_46typeOf))
, VAPTAG(useLabel(FN_NHC_46Internal_46_95apply1))
, VAPTAG(useLabel(FN_NonStdUnsafeCoerce_46unsafeCoerce))
, bytes2word(0,0,2,0)
, bytes2word(1,1,0,2)
, useLabel(CT_v516)
,};
Node FN_Prelude_46Show_46Data_46Dynamic_46Dynamic_46showsPrec[] = {
  bytes2word(ZAP_ARG_I1,NEEDSTACK_I16,PUSH_ZAP_ARG_I2,EVAL)
, bytes2word(NEEDHEAP_I32,UNPACK,2,HEAP_CVAL_I3)
, bytes2word(HEAP_CVAL_I4,HEAP_CVAL_I5,HEAP_OFF_N1,3)
, bytes2word(HEAP_OFF_N1,3,HEAP_CVAL_P1,6)
, bytes2word(HEAP_INT_P1,0,HEAP_P1,0)
, bytes2word(HEAP_CVAL_I3,HEAP_CVAL_P1,7,HEAP_CVAL_I5)
, bytes2word(HEAP_OFF_N1,3,HEAP_OFF_N1,3)
, bytes2word(HEAP_CVAL_P1,8,HEAP_OFF_N1,9)
, bytes2word(HEAP_OFF_N1,5,PUSH_HEAP,HEAP_CVAL_P1)
, bytes2word(8,HEAP_OFF_N1,15,HEAP_OFF_N1)
, bytes2word(5,RETURN_EVAL,ENDCODE,0)
, bytes2word(0,0,0,0)
,	/* CT_v516: (byte 0) */
  HW(6,2)
, 0
,};
Node F0_Prelude_46Show_46Data_46Dynamic_46Dynamic_46showsPrec[] = {
  CAPTAG(useLabel(FN_Prelude_46Show_46Data_46Dynamic_46Dynamic_46showsPrec),2)
, VAPTAG(useLabel(FN_Prelude_46showString))
, VAPTAG(useLabel(FN_LAMBDA463))
, VAPTAG(useLabel(FN_NHC_46Internal_46_95apply1))
, VAPTAG(useLabel(FN_Prelude_46Show_46Data_46Typeable_46TypeRep_46showsPrec))
, VAPTAG(useLabel(FN_LAMBDA464))
, VAPTAG(useLabel(FN_Prelude_46_46))
, bytes2word(0,0,0,0)
, useLabel(CT_v518)
,	/* FN_LAMBDA464: (byte 0) */
  bytes2word(NEEDHEAP_I32,PUSH_CADR_N1,2,STRING)
, bytes2word(RETURN,ENDCODE,0,0)
, bytes2word(0,0,0,0)
, CONSTRW(1,0)
, useLabel(ST_v517)
,	/* CT_v518: (byte 0) */
  HW(0,0)
, 0
,	/* CF_LAMBDA464: (byte 0) */
  VAPTAG(useLabel(FN_LAMBDA464))
, bytes2word(0,0,0,0)
, useLabel(CT_v520)
,	/* FN_LAMBDA463: (byte 0) */
  bytes2word(NEEDHEAP_I32,PUSH_CADR_N1,2,STRING)
, bytes2word(RETURN,ENDCODE,0,0)
, bytes2word(0,0,0,0)
, CONSTRW(1,0)
, useLabel(ST_v519)
,	/* CT_v520: (byte 0) */
  HW(0,0)
, 0
,	/* CF_LAMBDA463: (byte 0) */
  VAPTAG(useLabel(FN_LAMBDA463))
, bytes2word(1,0,0,1)
, useLabel(CT_v521)
,};
Node FN_Prelude_46Show_46Data_46Dynamic_46Dynamic_46show[] = {
  bytes2word(NEEDHEAP_I32,PUSH_HEAP,HEAP_CVAL_I3,HEAP_CVAL_I4)
, bytes2word(HEAP_ARG,1,RETURN_EVAL,ENDCODE)
, bytes2word(0,0,0,0)
,	/* CT_v521: (byte 0) */
  HW(2,1)
, 0
,};
Node F0_Prelude_46Show_46Data_46Dynamic_46Dynamic_46show[] = {
  CAPTAG(useLabel(FN_Prelude_46Show_46Data_46Dynamic_46Dynamic_46show),1)
, VAPTAG(useLabel(FN_Prelude_46_95_46show))
, useLabel(CF_Prelude_46Show_46Data_46Dynamic_46Dynamic)
, bytes2word(1,0,0,1)
, useLabel(CT_v522)
,};
Node FN_Prelude_46Show_46Data_46Dynamic_46Dynamic_46showsType[] = {
  bytes2word(NEEDHEAP_I32,PUSH_HEAP,HEAP_CVAL_I3,HEAP_CVAL_I4)
, bytes2word(HEAP_ARG,1,RETURN_EVAL,ENDCODE)
, bytes2word(0,0,0,0)
,	/* CT_v522: (byte 0) */
  HW(2,1)
, 0
,};
Node F0_Prelude_46Show_46Data_46Dynamic_46Dynamic_46showsType[] = {
  CAPTAG(useLabel(FN_Prelude_46Show_46Data_46Dynamic_46Dynamic_46showsType),1)
, VAPTAG(useLabel(FN_Prelude_46_95_46showsType))
, useLabel(CF_Prelude_46Show_46Data_46Dynamic_46Dynamic)
, bytes2word(1,0,0,1)
, useLabel(CT_v523)
,};
Node FN_Prelude_46Show_46Data_46Dynamic_46Dynamic_46showList[] = {
  bytes2word(NEEDHEAP_I32,PUSH_HEAP,HEAP_CVAL_I3,HEAP_CVAL_I4)
, bytes2word(HEAP_ARG,1,RETURN_EVAL,ENDCODE)
, bytes2word(0,0,0,0)
,	/* CT_v523: (byte 0) */
  HW(2,1)
, 0
,};
Node F0_Prelude_46Show_46Data_46Dynamic_46Dynamic_46showList[] = {
  CAPTAG(useLabel(FN_Prelude_46Show_46Data_46Dynamic_46Dynamic_46showList),1)
, VAPTAG(useLabel(FN_Prelude_46_95_46showList))
, useLabel(CF_Prelude_46Show_46Data_46Dynamic_46Dynamic)
, bytes2word(0,0,0,0)
, useLabel(CT_v524)
,	/* FN_Data_46Dynamic_46dynamicTc: (byte 0) */
  bytes2word(NEEDHEAP_I32,HEAP_CVAL_I3,PUSH_HEAP,HEAP_CVAL_I4)
, bytes2word(HEAP_OFF_N1,2,RETURN_EVAL,ENDCODE)
, bytes2word(0,0,0,0)
,	/* CT_v524: (byte 0) */
  HW(2,0)
, 0
,	/* CF_Data_46Dynamic_46dynamicTc: (byte 0) */
  VAPTAG(useLabel(FN_Data_46Dynamic_46dynamicTc))
, VAPTAG(useLabel(FN_LAMBDA465))
, VAPTAG(useLabel(FN_Data_46Typeable_46mkTyCon))
, bytes2word(0,0,0,0)
, useLabel(CT_v526)
,	/* FN_LAMBDA465: (byte 0) */
  bytes2word(NEEDHEAP_I32,PUSH_CADR_N1,2,STRING)
, bytes2word(RETURN,ENDCODE,0,0)
, bytes2word(0,0,0,0)
, CONSTRW(1,0)
, useLabel(ST_v525)
,	/* CT_v526: (byte 0) */
  HW(0,0)
, 0
,	/* CF_LAMBDA465: (byte 0) */
  VAPTAG(useLabel(FN_LAMBDA465))
, bytes2word(1,0,0,1)
, useLabel(CT_v527)
,};
Node FN_Data_46Typeable_46Typeable_46Data_46Dynamic_46Dynamic_46typeOf[] = {
  bytes2word(ZAP_ARG_I1,NEEDHEAP_I32,HEAP_CVAL_N1,1)
, bytes2word(PUSH_HEAP,HEAP_CVAL_I3,HEAP_CVAL_I4,HEAP_OFF_N1)
, bytes2word(3,RETURN_EVAL,ENDCODE,0)
, bytes2word(0,0,0,0)
, 0
, CONSTR(0,0,0)
,	/* CT_v527: (byte 0) */
  HW(2,1)
, 0
,};
Node F0_Data_46Typeable_46Typeable_46Data_46Dynamic_46Dynamic_46typeOf[] = {
  CAPTAG(useLabel(FN_Data_46Typeable_46Typeable_46Data_46Dynamic_46Dynamic_46typeOf),1)
, VAPTAG(useLabel(FN_Data_46Typeable_46mkTyConApp))
, useLabel(CF_Data_46Dynamic_46dynamicTc)
, bytes2word(0,0,0,0)
, useLabel(CT_v528)
,};
Node FN_Data_46Typeable_46Typeable_46Data_46Dynamic_46Dynamic[] = {
  bytes2word(NEEDHEAP_I32,PUSH_HEAP,HEAP_CVAL_N1,1)
, bytes2word(HEAP_CVAL_I3,RETURN,ENDCODE,0)
, bytes2word(0,0,0,0)
, 0
, CONSTR(0,1,0)
,	/* CT_v528: (byte 0) */
  HW(1,0)
, 0
,};
Node CF_Data_46Typeable_46Typeable_46Data_46Dynamic_46Dynamic[] = {
  VAPTAG(useLabel(FN_Data_46Typeable_46Typeable_46Data_46Dynamic_46Dynamic))
, useLabel(F0_Data_46Typeable_46Typeable_46Data_46Dynamic_46Dynamic_46typeOf)
, bytes2word(0,0,0,0)
, useLabel(CT_v529)
,};
Node FN_Prelude_46Show_46Data_46Dynamic_46Dynamic[] = {
  bytes2word(NEEDHEAP_I32,PUSH_HEAP,HEAP_CVAL_N1,1)
, bytes2word(HEAP_CVAL_I3,HEAP_CVAL_I4,HEAP_CVAL_I5,HEAP_CVAL_P1)
, bytes2word(6,RETURN,ENDCODE,0)
, bytes2word(0,0,0,0)
, 0
, CONSTR(0,4,0)
,	/* CT_v529: (byte 0) */
  HW(4,0)
, 0
,};
Node CF_Prelude_46Show_46Data_46Dynamic_46Dynamic[] = {
  VAPTAG(useLabel(FN_Prelude_46Show_46Data_46Dynamic_46Dynamic))
, useLabel(F0_Prelude_46Show_46Data_46Dynamic_46Dynamic_46showsPrec)
, useLabel(F0_Prelude_46Show_46Data_46Dynamic_46Dynamic_46showsType)
, useLabel(F0_Prelude_46Show_46Data_46Dynamic_46Dynamic_46showList)
, useLabel(F0_Prelude_46Show_46Data_46Dynamic_46Dynamic_46show)
,	/* ST_v484: (byte 0) */
  bytes2word(32,116,111,32)
, bytes2word(97,114,103,117)
, bytes2word(109,101,110,116)
,	/* ST_v519: (byte 2) */
  bytes2word(32,0,60,60)
,	/* ST_v517: (byte 1) */
 	/* ST_v486: (byte 4) */
  bytes2word(0,62,62,0)
, bytes2word(67,97,110,39)
, bytes2word(116,32,97,112)
, bytes2word(112,108,121,32)
, bytes2word(102,117,110,99)
, bytes2word(116,105,111,110)
,	/* ST_v498: (byte 2) */
  bytes2word(32,0,68,97)
, bytes2word(116,97,46,68)
, bytes2word(121,110,97,109)
, bytes2word(105,99,58,32)
, bytes2word(78,111,32,109)
, bytes2word(97,116,99,104)
, bytes2word(105,110,103,32)
, bytes2word(97,108,116,101)
, bytes2word(114,110,97,116)
, bytes2word(105,118,101,32)
, bytes2word(105,110,32,99)
, bytes2word(97,115,101,32)
, bytes2word(101,120,112,114)
, bytes2word(101,115,115,105)
, bytes2word(111,110,32,97)
, bytes2word(116,32,49,52)
, bytes2word(55,58,51,45)
, bytes2word(49,52,57,58)
,	/* ST_v510: (byte 4) */
  bytes2word(51,50,46,0)
, bytes2word(68,97,116,97)
, bytes2word(46,68,121,110)
, bytes2word(97,109,105,99)
, bytes2word(58,32,80,97)
, bytes2word(116,116,101,114)
, bytes2word(110,32,109,97)
, bytes2word(116,99,104,32)
, bytes2word(102,97,105,108)
, bytes2word(117,114,101,32)
, bytes2word(105,110,32,102)
, bytes2word(117,110,99,116)
, bytes2word(105,111,110,32)
, bytes2word(97,116,32,49)
, bytes2word(51,52,58,49)
, bytes2word(45,49,51,54)
, bytes2word(58,50,53,46)
,	/* ST_v525: (byte 1) */
  bytes2word(0,68,121,110)
, bytes2word(97,109,105,99)
,	/* ST_v488: (byte 1) */
  bytes2word(0,84,121,112)
, bytes2word(101,32,101,114)
, bytes2word(114,111,114,32)
, bytes2word(105,110,32,100)
, bytes2word(121,110,97,109)
, bytes2word(105,99,32,97)
, bytes2word(112,112,108,105)
, bytes2word(99,97,116,105)
, bytes2word(111,110,46,10)
, bytes2word(0,0,0,0)
,};
