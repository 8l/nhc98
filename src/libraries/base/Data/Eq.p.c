#include "newmacros.h"
#include "runtime.h"

#define ST_v155	((void*)startLabel+0)

static Node startLabel[] = {
 };
Node PM_Data_46Eq[] = {
 	/* ST_v155: (byte 0) */
  bytes2word(68,97,116,97)
, bytes2word(46,69,113,0)
,};
