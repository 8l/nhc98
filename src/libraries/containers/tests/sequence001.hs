
module Main where

import Data.Sequence

main :: IO ()
main = do print $ dropWhileL (< 3) $ fromList [1..5]
          print $ dropWhileR (> 3) $ fromList [1..5]

