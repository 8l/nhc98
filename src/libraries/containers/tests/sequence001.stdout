fromList [3,4,5]
fromList [1,2,3]
